doc/fuzzing.md

```diff
@@ -183,30 +181,26 @@ There are 3 ways fuzz tests can be built:
    tests would not be useful. This build is only useful for ensuring fuzz tests
    compile and link.
 
-## macOS hints for libFuzzer
+## macOS notes
 
-The default Clang/LLVM version supplied by Apple on macOS does not include
-fuzzing libraries, so macOS users will need to install a full version, for
-example using `brew install llvm`.
+Support for fuzzing on macOS is not officially maintained by this project. If
+you are running into issues on macOS, we recommend fuzzing on Linux instead for
+best results. On macOS this can be done within Docker or a virtual machine.
 
-You may also need to take care of giving the correct path for `clang` and
-`clang++`, like `CC=/path/to/clang CXX=/path/to/clang++` if the non-systems
-`clang` does not come first in your path.
+Reproducing and debugging fuzz testcases on macOS is supported, by building the
+fuzz binary without support for any specific fuzzing engine.
 
-Using `lld` is required due to issues with Apple's `ld` and `LLVM`.
+You may still be able to fuzz on macOS using the following steps (provided on
+best effort basis, only aimed at the latest macOS version, may be outdated):
 
-Full configuration step for macOS:
-
-```sh
+```
 $ brew install llvm lld
-$ cmake --preset=libfuzzer \
+$ cmake --preset=libfuzzer-nosan \
```

Comment:
As described in https://github.com/bitcoin/bitcoin/pull/32084#issuecomment-3361478681 this doesn't seem to be enough for me, I'm getting a lot of failures locally on my non-intel-based Mac.
It needs a few more parameters to avoid being tangled with local `AppleClang`:
```
cmake --preset=libfuzzer-nosan \
   -DCMAKE_C_COMPILER="$(brew --prefix llvm)/bin/clang" \
   -DCMAKE_CXX_COMPILER="$(brew --prefix llvm)/bin/clang++" \
   -DCMAKE_OSX_SYSROOT="$(xcrun --show-sdk-path)" \
   -DCMAKE_EXE_LINKER_FLAGS="-fuse-ld=lld -L$(brew --prefix llvm)/lib/c++"
```

Reply:
> This comment led me to remove the steps entirely

Hmmm, so what happened to "best effort"?

> which is partly what we're trying to avoid with this effort in the first place

I thought we were aiming to provide Mac users a way to have at least basic fuzzing support.

Reply:
> it depends on macOS version, brew version etc

What if we documented a solution that would work with latest xcode and clang and whatever else is needed?
That shouldn't be too difficult and shouldn't involve "bike shedding" (I don't think it's the correct term here since it's not a "minor, easily understandable issues" at the expense of "neglecting more complex, important matters").
 
As mentioned, I strongly think we should have guidance for at least basic fuzzing support for Mac as well!